internal/sbi/producer/ue_context.go

```diff
@@ -336,11 +376,64 @@ func UEContextTransferProcedure(ueContextID string, ueContextTransferRequest mod
 	return ueContextTransferResponse, nil
 }
 
-func buildUEContextModel(ue *context.AmfUe) *models.UeContext {
+func buildUEContextModel(ue *context.AmfUe, integrityProtected bool) *models.UeContext {
```

Comment:
According to TS 29.518 5.2.2.2.1.1
> 2a. On success: 
> - if the reason attribute is "INIT_REG" and integrity check is successful, the (source) AMF shall respond with 
the status code "200 OK". The payload of the response shall be an object of "UeContextTransferRspData" 
data type, containing: 
> case a) the representation of the requested UE Context without PDU Session Contexts; or 
> case b) the representation of the requested UE Context only containing the "supi" attribute, if the UE is 
registered in a different access type in the (source) AMF and the source AMF determines based on the 
PLMN ID of the (target) AMF that there is no possibility for relocating the N2 interface to the (target) 
AMF. 
> - If the reason attribute is "MOBI_REG" and integrity check is successful, the (source) AMF shall respond 
with the status code "200 OK". The payload of the response shall be an object of 
"UeContextTransferRspData" data type, containing the representation of the complete UE Context including 
available PDU Session Contexts. 

and 5.2.2.2.1.2 
> 2. Same as step 2a of figure 5.2.2.2.1.1-1, with following differences: 
> - The (source) AMF shall skip integrity check and shall respond with the status code "200 OK "with the UE 
Context excluding SeafData and including available PDU Session Contexts

I think transfer reason is more suitable than the `integrityProtected` to control the items to be filled in here. If you pass the transfer reason instead of the `integrityProtected` in, you can even move the `sessionContextList` parts in `case models.TransferReason_MOBI_REG_UE_VALIDATED` into this api.